Training/Medium/gift.rb

```diff
@@ -0,0 +1,42 @@
+# Solution for https://www.codingame.com/ide/puzzle/the-gift
+class Gift
+  attr_accessor :participants, :gift_price, :budgets_list
+
+  def initialize
+    @participants = gets.to_i
+    @gift_price = gets.to_i
+    @budgets_list = init_budgets_list
+  end
+
+  def init_budgets_list
+    list = []
+    participants.times do
+      list << gets.to_i
+    end
+    list.sort
+  end
+
+  def not_possible?
```

Comment:
Methods should have one purpose. This one tells us if something is possible - true/false. That said, it will look something like:
```ruby
budgets_list.inject(0, :+) < gift_price
```